packages/components/forms/src/text-input/types.ts

```diff
@@ -0,0 +1,22 @@
+import { BaseInputInternalProps } from '../base-input/types';
+
+export interface TextInputProps extends Omit<BaseInputInternalProps, 'as'> {
+  /**
+   * Controlls the character count for the input, works together with maxLength propery
+   * @default false
+   */
+  countCharacters?: boolean;
+  /**
+   * Pass max number of characters for the input, needed for countCharacters prop
+   */
+  maxLength?: number;
```

Comment:
I think this is also a native `input` attribute, so maybe no need to put it in here 🤷 

that's why we also need `countCharacters` to tell if we should show the counter or not